cli/build.rs

```diff
@@ -41,21 +44,51 @@ fn main() -> std::io::Result<()> {
     }
     println!("cargo:rerun-if-env-changed=NIX_JJ_GIT_HASH");
 
-    if let Some(git_hash) = get_git_hash() {
-        println!("cargo:rustc-env=JJ_VERSION={}-{}", version, git_hash);
+    if let Some((mut git_hash, maybe_date)) = get_git_timestamp_and_hash() {
+        git_hash.truncate(16);
+        println!(
+            "cargo:rustc-env=JJ_VERSION={}+{}-{}",
+            version,
```

Comment:
unrelated to this PR, but it would be simpler to use `CARGO_PKG_VERSION` instead of parsing the version from `cargo metadata` https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts (see "CARGO_PKG_<var>")